solana-programs/Dockerfile.dev

```diff
@@ -1,139 +1,83 @@
-# compilation.
-RUN --mount=type=cache,id=anchor-build-cache,target=/usr/src/app/payment-router/target \
-    --mount=type=cache,id=anchor-solana-cache,target=/root/.cache/solana \
-    --mount=type=cache,id=anchor-cargo-registry,target=/usr/local/cargo/registry \
-    mkdir -p artifacts/anchor-target && \
-    ./scripts/update-keys.sh && \
-    cd payment-router && anchor build && cd .. && \
-    cp -r payment-router/target/deploy artifacts/anchor-target && \
-    cp -r payment-router/target/idl artifacts/anchor-target && \
-    cp -r payment-router/target/types artifacts/anchor-target
-
-# ------
-FROM rust:1.69.0
-
-WORKDIR /usr/src/app
-
-ENV CARGO_INCREMENTAL=1
-ENV NODE_MAJOR=18
-RUN apt-get update && \
-    apt-get install -y ca-certificates curl gnupg && \
-    mkdir -p /etc/apt/keyrings && \
-    curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg && \
-    echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list && \
-    apt-get update && \
-    apt-get install -y clang cmake build-essential libudev-dev libhidapi-dev pkg-config libssl-dev git nodejs && \
-    npm install -g yarn @coral-xyz/anchor-cli@0.28.0
-
-ENV PATH="/root/solana-release/bin:${PATH}"
+    apt-get install -y \
+        build-essential \
+        pkg-config \
+        libudev-dev llvm libclang-dev \
+        protobuf-compiler libssl-dev
+
+# Installs Solana CLIs either from the install script (for available builds like linux x86_64)
+# or using cargo install to build from source (for unavailable builds like linux aarch64)
 COPY scripts/install-solana.sh /tmp/install-solana.sh
-RUN --mount=type=cache,id=deploy-cargo-cache,target=/usr/local/cargo/registry /tmp/install-solana.sh
-
-ARG AUDIUS_ETH_REGISTRY_PRIVATE_KEY
-ARG TRACK_LISTEN_COUNT_PRIVATE_KEY
-ARG CLAIMABLE_TOKENS_PRIVATE_KEY
-ARG REWARD_MANAGER_PRIVATE_KEY
-ARG AUDIUS_DATA_PRIVATE_KEY
-ARG PAYMENT_ROUTER_PRIVATE_KEY
-ARG owner_private_key
-ARG feepayer_private_key
-ARG token_private_key
-ARG fake_usdc_token_private_key
-ARG admin_authority_private_key
-ARG admin_account_private_key
-ARG signer_group_private_key
-ARG valid_signer_private_key
-ARG reward_manager_pda_private_key
-ARG reward_manager_token_pda_private_key
-ARG valid_signer_eth_address
-
-COPY . .
-COPY --from=bpf-builder /usr/src/app/artifacts/main-target target
-COPY --from=anchor-builder /usr/src/app/artifacts/anchor-target/. target
-
-RUN --mount=type=cache,id=deploy-solana-cache,target=/root/.cache/solana \
-    --mount=type=cache,id=deploy-cargo-cache,target=/usr/local/cargo/registry \
-    ./scripts/setup-predeployed.sh /usr/db
+RUN --mount=type=cache,id=cargo-cache,target=/usr/local/cargo/registry  /tmp/install-solana.sh
+ENV PATH="/root/solana-release/bin:${PATH}"
 
+# Ensures the test validator is running
 HEALTHCHECK --interval=5s --timeout=5s --retries=10 \
```

Comment:
change this to
```
HEALTHCHECK --interval=5s --timeout=5s --retries=10 \
    CMD curl -f http://localhost:8899/health || exit 1
```

because in install-solana, we may only get the test validator